### DISCUSSION TITLE
[LiveComponent] LiveProps values at template rendering

### DISCUSSION THREAD

**Discussion by fGuix** [symfony/ux#2920] [Questions & Answers]
Hi,

I'm using live component in multiple projects and I face now a case that confuses me.
In my case, I have some pagination values (first, last, next, previous) for a list (my component) of elements.
The initial value defined in my component: 
``` php
#[LiveProp]
public int $currentPage = 1;
#[LiveProp]
public ?int $first = null;
#[LiveProp]
public ?int $previous = null;
#[LiveProp]
public ?int $next = null;
#[LiveProp]
public ?int $last = null;
```

I have a method that changes those values in component init (thanks to ExposeInTemplate)
```php
  #[ExposeInTemplate]
  public function initLessons(): void
  {
    ... // adapting values
  }
```

Now, take a look at this dump I triggered in the component template:
<img width="455" alt="Capture d’écran 2024-04-22 à 12 38 34" src="https://github.com/symfony/ux/assets/81959747/1d4ed1db-94fc-4893-a354-07d20bde1ce0">

We can see the values accessed through 'this' have been updated but not the values at the root ("liveprops" ?).

Is it an expected behaviour ?

**Comment by smnandre** [comment]
> I have a method that changes those values in component init (thanks to ExposeInTemplate)

The `ExposeInTemplate` attribute is not meant to be used on void method, but on properties or getters 

What is the problem you wanted to solve using this attribute initially ?

**Comment by fGuix** [comment]
I'm typically using the #[ExposeInTemplate] attribute to initialize values before rendering component. In this case, it's getting the list of items and initializing the pagination values used in the frontend template.

**Comment by smnandre** [comment]
Yeah but this is not the goal of the ExposeInTemplate attribute, and the documentation is pretty clear about what it does.. 

To be honest i'm even surprised it does not throw an error in your case, and this is something that could happen anytime.. 

To initialize things you would have better / stable results using the dedicated PreMount / PostMount hooks

**Comment by fGuix** [comment]
ok. Thanks for the pointers. I will change accordingly and see the behavior.

**Comment by fGuix** [comment]
all good, thanks for the help.
